client/src/navigator/Drawer/SplitDrawerItemList.js

```diff
@@ -0,0 +1,159 @@
+ * @param descriptors - Contains styling information for the Drawer.Item
+ * @param firstRouteNames - contains the names of the routes that must appear in the first navigation list
+ * @param secondRouteNames - contains the names of the routes that must appear in the second navigation list
+ * @param middleChildComponent - the child component that should be rendered in between two navigation lists
+ *
+ * Original component: https://github.com/react-navigation/react-navigation/blob/main/packages/drawer/src/views/DrawerItemList.tsx
+ */
+const SplitDrawerItemList = ({
+  state,
+  navigation,
+  descriptors,
+  firstRouteNames,
+  secondRouteNames,
+  middleChildComponent,
+}) => {
+  const buildLink = useLinkBuilder()
+
+  const focusedRoute = state.routes[state.index]
+  const focusedDescriptor = descriptors[focusedRoute.key]
+  const focusedOptions = focusedDescriptor.options
+
+  console.log(focusedOptions)
+
+  /**
+   * Creates a DrawerItem given route data
+   * @param route = The route data
+   * @param i - The index of the route in the React Navigator
+   * @param zeroBasedShift - A shift to apply in the instance the route belongs to the second list
+   * @returns
+   */
+  const createCustomDrawerItem = (route, i, zeroBasedShift = 0) => {
+    const focused = i === state.index - zeroBasedShift
+
+    const {
+      activeBackgroundColor,
+      activeTintColor,
+      inactiveBackgroundColor,
+      inactiveTintColor,
+    } = focusedOptions
+
+    const onPress = () => {
+      const event = navigation.emit({
+        type: 'drawerItemPress',
+        target: route.key,
+        canPreventDefault: true,
+      })
+
+      if (!event.defaultPrevented) {
+        navigation.dispatch({
+          ...(focused
+            ? DrawerActions.closeDrawer()
+            : CommonActions.navigate({ name: route.name, merge: true })),
+          target: state.key,
+        })
+      }
+    }
+
+    const {
+      title,
+      drawerLabel,
+      drawerIcon,
+      drawerLabelStyle,
+      drawerItemStyle,
+      drawerAllowFontScaling,
+    } = descriptors[route.key].options
+
+    let label = drawerLabel
+
+    if (drawerLabel === undefined) {
+      label = title !== undefined ? title : route.name
+    }
+
+    return (
+      <DrawerItem
+        key={route.key}
+        label={label}
+        icon={drawerIcon}
+        focused={focused}
+        activeTintColor={activeTintColor}
+        inactiveTintColor={inactiveTintColor}
+        activeBackgroundColor={activeBackgroundColor}
+        inactiveBackgroundColor={inactiveBackgroundColor}
+        allowFontScaling={drawerAllowFontScaling}
+        labelStyle={drawerLabelStyle}
+        style={drawerItemStyle}
+        to={buildLink(route.name, route.params)}
+        onPress={onPress}
+      />
+    )
+  }
+
+  // Create the two Drawer.Item lists
+  const firstRoutesList = state.routes
+    .filter((obj) => firstRouteNames.includes(obj.name))
```

Comment:
more specific names if this component wont be reused

Reply:
I'm going to keep the general name in the event we reuse the component in the future.